src/test/lib/OwnershipTest.cpp

```diff
@@ -230,3 +230,44 @@ void QuicTestRegistrationShutdownAfterConnOpenAndStart()
     TEST_TRUE(State.StateEvent.WaitTimeout(2000));
     TEST_EQUAL(1, State.ShutdownCount);
 }
+
+void QuicTestConnectionCloseBeforeStreamClose()
+{
+    MsQuicRegistration Registration;
+    TEST_TRUE(Registration.IsValid());
+
+    MsQuicRegistration ServerRegistration{true};
+    TEST_TRUE(ServerRegistration.IsValid());
+
+    MsQuicAlpn Alpn("MsQuicTest");
+    MsQuicCredentialConfig ClientCredConfig;
+    MsQuicConfiguration ClientConfiguration(Registration, Alpn, ClientCredConfig);
+    TEST_QUIC_SUCCEEDED(ClientConfiguration.GetInitStatus());
+
+    MsQuicConfiguration ServerConfiguration(ServerRegistration, Alpn, ServerSelfSignedCredConfig);
+    TEST_QUIC_SUCCEEDED(ServerConfiguration.GetInitStatus());
+
+    MsQuicAutoAcceptListener Listener(ServerRegistration, ServerConfiguration, MsQuicConnection::NoOpCallback);
+    TEST_QUIC_SUCCEEDED(Listener.GetInitStatus());
+    QUIC_ADDRESS_FAMILY QuicAddrFamily = QUIC_ADDRESS_FAMILY_INET;
+    QuicAddr ServerLocalAddr(QuicAddrFamily);
+    TEST_QUIC_SUCCEEDED(Listener.Start(Alpn, &ServerLocalAddr.SockAddr));
+    TEST_QUIC_SUCCEEDED(Listener.GetLocalAddr(ServerLocalAddr));
+
+    {
+        MsQuicConnection Connection(Registration);
+        TEST_QUIC_SUCCEEDED(Connection.GetInitStatus());
+        MsQuicStream Stream(Connection, QUIC_STREAM_OPEN_FLAG_NONE);
+        TEST_QUIC_SUCCEEDED(Stream.GetInitStatus());
+        Connection.Close();
+    }
```

Comment:
Just a quick question regarding the testing methodology. The way I understand this change is, we want apps to still be able to make API calls to Streams after connection closure without getting a seg-fault. Shouldn't `Connection.close()` here be before we call `Stream.GetInitStatus()`?

Reply:
This makes it so the stream is closed after the connection. We could also add additional calls on the stream after this connection close call, for instance to query the stream ID. I can add that for extra coverage.

Reply:
Ahh, Streams implicitly auto-close. Before, apps had to explicitly close all streams before connection closure. Yeah sounds good, adding additional queries to Streams after connection closure ensures no seg-faults occur.